src/vm/hints/hint_utils.rs

```diff
@@ -792,6 +792,165 @@ pub fn sqrt(
+        Some(range_check_ptr_ref),
+        Some(div_ref),
+        Some(value_ref),
+        Some(bound_ref),
+    ) = (
+        ids.get(&String::from("r")),
+        ids.get(&String::from("biased_q")),
+        ids.get(&String::from("range_check_ptr")),
+        ids.get(&String::from("div")),
+        ids.get(&String::from("value")),
+        ids.get(&String::from("bound")),
+    ) {
+        (
+            r_ref,
+            biased_q_ref,
+            range_check_ptr_ref,
+            div_ref,
+            value_ref,
+            bound_ref,
+        )
+    } else {
+        return Err(VirtualMachineError::IncorrectIds(
+            vec![
+                String::from("r"),
+                String::from("biased_q"),
+                String::from("range_check_ptr"),
+                String::from("div"),
+                String::from("value"),
+                String::from("bound"),
+            ],
+            ids.into_keys().collect(),
+        ));
+    };
+    //Check that each reference id corresponds to a value in the reference manager
+    let (r_addr, biased_q_addr, range_check_ptr_addr, div_addr, value_addr, bound_addr) = if let (
+        Some(r_addr),
+        Some(biased_q_addr),
+        Some(range_check_ptr_addr),
+        Some(div_addr),
+        Some(value_addr),
+        Some(bound_addr),
+    ) = (
+        get_address_from_reference(r_ref, &vm.references, &vm.run_context, vm),
+        get_address_from_reference(biased_q_ref, &vm.references, &vm.run_context, vm),
+        get_address_from_reference(range_check_ptr_ref, &vm.references, &vm.run_context, vm),
+        get_address_from_reference(div_ref, &vm.references, &vm.run_context, vm),
+        get_address_from_reference(value_ref, &vm.references, &vm.run_context, vm),
+        get_address_from_reference(bound_ref, &vm.references, &vm.run_context, vm),
+    ) {
+        (
+            r_addr,
+            biased_q_addr,
+            range_check_ptr_addr,
+            div_addr,
+            value_addr,
+            bound_addr,
+        )
+    } else {
+        return Err(VirtualMachineError::FailedToGetIds);
+    };
+    match (
+        vm.memory.get(&r_addr),
+        vm.memory.get(&biased_q_addr),
+        vm.memory.get(&range_check_ptr_addr),
+        vm.memory.get(&div_addr),
+        vm.memory.get(&value_addr),
+        vm.memory.get(&bound_addr),
+    ) {
+        (
+            Ok(_),
+            Ok(_),
+            Ok(_),
+            Ok(Some(maybe_rel_div)),
+            Ok(Some(maybe_rel_value)),
+            Ok(Some(maybe_rel_bound)),
+        ) => {
+            for (name, builtin) in &vm.builtin_runners {
+                //Check that range_check_builtin is present
+                let builtin = match builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>() {
+                    Some(b) => b,
+                    None => return Err(VirtualMachineError::NoRangeCheckBuiltin),
+                };
+
+                if name == &String::from("range_check") {
```

Comment:
Considering you're attempting the cast regardless of name, all this could be:
```suggestion
                if let Some(builtin) = builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>() {
```

Reply:
resolved https://github.com/lambdaclass/cleopatra_cairo/pull/279/commits/89e182f2098f0923837b87b6553ad00628ce87fd